### PR TITLE
[pavlo_turchyn]Task 11 20

### PR DESCRIPTION
**Pull request by FabianMoretti**

### CODE REVIEW

**Comment by prezlyataa** [comment]
You have conflicts here, should resolve it.

File: tasks/12_title-case/title-case.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* eslint "require-jsdoc": 0 */
+
+export const titleCase = (str) => {
+  let lowStr = str.toLowerCase();
```
Comment (tarasbobak) [inline tasks/12_title-case/title-case.js]: You need to understand the difference between let and const. Const is used for values that will not be reassigned. Let on the other hand is used for variables which will be reassigned

File: tasks/12_title-case/title-case.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  let lowArr = lowStr.split(' ');
+  let resArr = [];
+  let lowArrLength = lowArr.length;
+  for (let i = 0; i < lowArrLength; i++) {
+    resArr.push(lowArr[i][0].toUpperCase() +
```
Comment (tarasbobak) [inline tasks/12_title-case/title-case.js]: I don't like the way you wrapped this line
You could go one line:
resArr.push(lowArr[i][0].toUpperCase() + lowArr[i].slice(1));
Or make variable for first uppercase value and reused it
Please read about line wrapping rules:
https://google.github.io/styleguide/jsguide.html#formatting-line-wrapping

File: tasks/13_largest-of-four/largest-of-four.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* eslint "require-jsdoc": 0 */
+
+export const largestOfFour = (bigArr) => {
+  let maxArr = [];
+  for (let i = 0; i < bigArr.length; i++) {
+    maxArr.push(Math.max.apply(null, bigArr[i]));
```
Comment (tarasbobak) [inline tasks/13_largest-of-four/largest-of-four.js]: nice

File: tasks/14_clone-arr/clone-arr.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/* eslint "require-jsdoc": 0 */
+
+export const cloneArr = (original) => {
+  return original.concat();
```
Comment (tarasbobak) [inline tasks/14_clone-arr/clone-arr.js]: interesting 👍

File: tasks/15_confirm-ending/confirm-ending.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+/* eslint "require-jsdoc": 0 */
+
+export const confirmEnding = (str, target) => {
+  return str.substr(-target.length)===target;
```
Comment (tarasbobak) [inline tasks/15_confirm-ending/confirm-ending.js]: add whitespace between ===, like `return str.substr(-target.length) === target;`

File: tasks/16_repeat-string/repeat-string.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/* eslint "require-jsdoc": 0 */
+
+export const repeatStringNumTimes = (str, counter) => {
+  let res = '';
+  for (let i = 0; i < counter; i++) {
+    res+=str;
```
Comment (tarasbobak) [inline tasks/16_repeat-string/repeat-string.js]: res += str;

File: tasks/17_truncate-string/truncate-string.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+/* eslint "require-jsdoc": 0 */
+
+export const truncateString = (str, n) => {
+  if (n>=str.length) return str;
```
Comment (tarasbobak) [inline tasks/17_truncate-string/truncate-string.js]: if (n >= str.length)
if (n > 3)

File: tasks/18_humanize-format/humanize-format.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+/* eslint "require-jsdoc": 0 */
+
+export const humanizeFormat = (num) => {
+  let strNum = num + '';
+  let lastStrNum =strNum[strNum.length - 1];
```
Comment (tarasbobak) [inline tasks/18_humanize-format/humanize-format.js]: it's good that you cached value, but it has to be const first.
Secondly you could make number out of string once and not each time, like:
const ending = parseInt(strNum[strNum.length - 1], 10);

File: tasks/19_array-chunks/array-chunks.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+/* eslint "require-jsdoc": 0 */
+
+export const arrayChunks = (mainArr, size) => {
+  let result = [];
+  let i = 0;
+  let n = mainArr.length;
```
Comment (tarasbobak) [inline tasks/19_array-chunks/array-chunks.js]: n does not tell a lot to a reader. Consider better naming. like arrLength

File: tasks/20_truncate-array/truncate-array.js [hunk 10]
```diff
@@ -0,0 +1,5 @@
+/* eslint "require-jsdoc": 0 */
+export const truncateArray = (arr, size) => {
+  arr.splice(0, size);
```
Comment (tarasbobak) [inline tasks/20_truncate-array/truncate-array.js]: interesting, but generally it is not a good practice to change incoming data. So ideally you should not change the value for arr itself but return a new array.

**Comment by tarasbobak** [comment]
Good job. Please take into account comments for future work!

**Pull request merged**